src/Workspaces/Core/Portable/Workspace/Workspace_Events.cs

```diff
@@ -59,14 +77,25 @@ protected Task RaiseWorkspaceChangedEventAsync(WorkspaceChangeKind kind, Solutio
             projectId = documentId.ProjectId;
         }
 
-        var ev = GetEventHandlers<WorkspaceChangeEventArgs>(WorkspaceChangeEventName);
+        var ev = GetEventHandlers<WorkspaceChangeEventArgs>(WorkspaceChangedImmediateEventName);
+        WorkspaceChangeEventArgs args = null;
+        if (ev.HasHandlers)
+        {
+            using (Logger.LogBlock(FunctionId.Workspace_EventsImmediate, (s, p, d, k) => $"{s.Id} - {p} - {d} {kind.ToString()}", newSolution, projectId, documentId, kind, CancellationToken.None))
+            {
+                args = new WorkspaceChangeEventArgs(kind, oldSolution, newSolution, projectId, documentId);
+                ev.RaiseEvent(static (handler, arg) => handler(arg.self, arg.args), (self: this, args));
```

Comment:
aside, are these exception safe if the attached handler throws an exception?  i ask because the old system might have been ok ift he Task went into a faulted state.  i'm not sure about the new system.

Reply:
Oops asked this same question here: https://github.com/dotnet/roslyn/pull/76932#discussion_r1931416071 I think this might cause an issue now if this were to throw so probably best to handle that better.

Reply:
Any concern with me just changing RaiseEvent from using FatalError.ReportAndPropagate to instead use FatalError.ReportAndCatch (also, would it make sense to have the try/catch inside the foreach in that method?)

Reply:
@jasonmalinowski -- any thoughts on change RaiseEvent to instead use FatalError.ReportAndCatch (and to move the try/catch inside the loop)?

Reply:
I went ahead and just made the change to switch to ReportAndCatch